gocourse02/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import "fmt"
+
+type Animal struct {
+	name    string
+	species string
+	cage    *Cage
+}
+
+type Zoo struct {
+	Animals []*Animal
+}
+
+type Cage struct {
+	name   string
+	animal *Animal
+}
+
+type AllCages struct {
+	Cages []Cage
+}
+
+type Zookeeper struct {
+	name string
+}
+
+func NewAnimal(species string, name string) *Animal {
+	return &Animal{species: species, name: name}
+}
+func (a *Animal) Escape() {
+	fmt.Printf("Animal %s escaped from the cage %v\n", a.name, a.cage.name)
+	a.cage.animal = nil
+	a.cage = nil
+}
+func NewCage(name string) *Cage {
```

Comment:
Між методами, структурами та функціями краще додавати пустий рядок для полегшення читання:

```suggestion
}

func (a *Animal) Escape() {
	fmt.Printf("Animal %s escaped from the cage %v\n", a.name, a.cage.name)
	a.cage.animal = nil
	a.cage = nil
}

func NewCage(name string) *Cage {
```

Інколи методи не розділяються пустими рядками. Але це короткі однорядкові методи, які рідко зустрінеш. Наприклад, https://github.com/golang/go/blob/d36353499f673c89a267a489beb80133a14a75f9/src/cmd/compile/internal/ir/func.go#L188-L192